### ISSUE TITLE
Ctrl-C in command substitution exits parent shell

### ISSUE THREAD

**Issue by jyn514** [oilshell/oil#467]
Repro steps:

```
(bash) ~ $ osh --rcfile /dev/null
osh$ `sleep 1`
^C
(bash) ~ $
```

Expected behavior: osh sends interrupt to command substitution, but continues reading input and behaving as an interactive shell (same as bash and dash)

Actual behavior: osh exits completely!

This looks like it was introduced by https://github.com/oilshell/oil/commit/119123831356b97b6557e3dd186cb28ec69c7f4a, which enables 'Ctrl-C' during command substitions. osh should distinguish between interrupting the substitution and interrupting the parent shell.

**Comment by andychu** [comment]
OK I reproduced this... my memory of how it works is already fuzzy :-/

But thank you for the report and testing!

**Comment by andychu** [comment]
Hm looks like this fell through the cracks, should take another look at it

**Comment by andychu** [comment]
Fixed in 0.9.7: https://www.oilshell.org/blog/2022/01/notes-themes.html

As mentioned in the blog post, I probably subconsciously put this off because we didn't have a good test harness!
